cfg/c64-overlay.cfg

```diff
@@ -1,64 +1,70 @@
+    OVL6:     file = "%O.6",           start = __HIMEM__  - __OVERLAYSIZE__, size = __OVERLAYSIZE__;
+    OVL7ADDR: file = "%O.7",           start = __HIMEM2__ - __OVERLAYSIZE__, size = $0002;
+    OVL7:     file = "%O.7",           start = __HIMEM__  - __OVERLAYSIZE__, size = __OVERLAYSIZE__;
+    OVL8ADDR: file = "%O.8",           start = __HIMEM2__ - __OVERLAYSIZE__, size = $0002;
+    OVL8:     file = "%O.8",           start = __HIMEM__  - __OVERLAYSIZE__, size = __OVERLAYSIZE__;
+    OVL9ADDR: file = "%O.9",           start = __HIMEM2__ - __OVERLAYSIZE__, size = $0002;
+    OVL9:     file = "%O.9",           start = __HIMEM__  - __OVERLAYSIZE__, size = __OVERLAYSIZE__;
 }
 SEGMENTS {
-    LOADADDR: load = LOADADDR, type = ro;
-    EXEHDR:   load = HEADER,   type = ro;
-    STARTUP:  load = RAM,      type = ro;
-    LOWCODE:  load = RAM,      type = ro,                optional = yes;
-    INIT:     load = RAM,      type = ro,  define = yes, optional = yes;
-    CODE:     load = RAM,      type = ro;
-    RODATA:   load = RAM,      type = ro;
-    DATA:     load = RAM,      type = rw;
-    ZPSAVE:   load = RAM,      type = bss;
-    BSS:      load = RAM,      type = bss, define = yes;
-    ZEROPAGE: load = ZP,       type = zp;
-    OVL1ADDR: load = OVL1ADDR, type = ro;
-    OVERLAY1: load = OVL1,     type = ro,  define = yes, optional = yes;
-    OVL2ADDR: load = OVL2ADDR, type = ro;
-    OVERLAY2: load = OVL2,     type = ro,  define = yes, optional = yes;
-    OVL3ADDR: load = OVL3ADDR, type = ro;
-    OVERLAY3: load = OVL3,     type = ro,  define = yes, optional = yes;
-    OVL4ADDR: load = OVL4ADDR, type = ro;
-    OVERLAY4: load = OVL4,     type = ro,  define = yes, optional = yes;
-    OVL5ADDR: load = OVL5ADDR, type = ro;
-    OVERLAY5: load = OVL5,     type = ro,  define = yes, optional = yes;
-    OVL6ADDR: load = OVL6ADDR, type = ro;
-    OVERLAY6: load = OVL6,     type = ro,  define = yes, optional = yes;
-    OVL7ADDR: load = OVL7ADDR, type = ro;
-    OVERLAY7: load = OVL7,     type = ro,  define = yes, optional = yes;
-    OVL8ADDR: load = OVL8ADDR, type = ro;
-    OVERLAY8: load = OVL8,     type = ro,  define = yes, optional = yes;
-    OVL9ADDR: load = OVL9ADDR, type = ro;
-    OVERLAY9: load = OVL9,     type = ro,  define = yes, optional = yes;
+    ZEROPAGE: load = ZP,              type = zp;
+    LOADADDR: load = LOADADDR,        type = ro;
+    EXEHDR:   load = HEADER,          type = ro;
+    STARTUP:  load = RAM,             type = ro;
+    LOWCODE:  load = RAM,             type = ro,                optional = yes;
+    CODE:     load = RAM,             type = ro;
+    RODATA:   load = RAM,             type = ro;
+    DATA:     load = RAM,             type = rw;
+    ZPSAVE:   load = RAM,             type = bss, define = yes;
+    BSS:      load = RAM,             type = bss, define = yes;
+    INIT:     load = MOVE, run = RAM, type = ro,  define = yes, optional = yes;
```

Comment:
Given that the startup code itself uses the INIT segment it doesn't have to be `optional`. (Yes: This is "wrong" in the Apple config files too). Another side effect is that the routine to copy the INIT segment doesn't need to take care of an empty segment.